papers/extract.py

```diff
@@ -20,55 +20,46 @@
 # PDF parsing / crossref requests
 # ===============================
 
-def createrandname(path, ext):
-    import random
-    import string
-    check = False
-    while check == False:
-        rand = "".join(
-            random.sample(string.ascii_lowercase + string.digits, 20)
-        ) + ext
-        check = True if not os.path.exists(path + rand) else False
-    return rand
-
-
 def readpdf(pdf, first=None, last=None, keeptxt=False):
     import shutil
-    txtfile = pdf.replace('.pdf', '.txt')
-    imgfile = pdf.replace('.pdf', '.png')
-    # txtfile = os.path.join(os.path.dirname(pdf), pdf.replace('.pdf','.txt'))
+    import tempfile
     if True:  # not os.path.exists(txtfile):
-        # logger.info(' '.join(['pdftotext','"'+pdf+'"', '"'+txtfile+'"']))
         if os.path.isfile(pdf):
             path, ext = os.path.splitext(pdf)
-            path = "/".join(path.split("/")[:-1])
-            rand = createrandname(path, ext)
-            # Do the rename
-            newfile = os.path.join(path, rand)
-            shutil.copyfile(pdf, newfile)
-            logger.info("\t->", rand)
-            pdf = newfile
+            path = os.path.dirname(path)
+            fd, uniq_pdf = tempfile.mkstemp(dir=path, suffix=ext)
```

Comment:
That's all right for now, but eventually we will want to move all that in a temporary directory somewhere. It is not good to create files in a user-provided directory. Note it was already the case with my original version, but you make me aware of this now.

Reply:
Ok, changing the following will make those files in `/tmp`
```suggestion
            fd, uniq_pdf = tempfile.mkstemp(suffix=ext)
```